packages/runtime/src/runtime.ts

```diff
@@ -461,9 +466,15 @@ export class Runtime {
     let mode = ExecutionMode.STATEFUL;
     if (!Array.isArray(txnParams)) {
       mode = this.getExecutionMode(txnParams);
+      if (txnParams.sign === SignType.LogicSignature) {
+        this.validateLsig(txnParams);
```

Comment:
We have `run` in line 483 as well. This way we would run stateless teal code twice. Right?

Reply:
No, because program passed is different.

Reply:
Ok, so that means `program` in `executeTx` can be optional (if the tx or txGroup contains only stateless txns). We may need to update this is well. We can do it in the group txns task

Reply:
exactly, also we don't need this for payment only too.
we need to restructure it.